src/components/dialog/index.js

```diff
@@ -137,7 +137,13 @@ class Dialog extends BaseComponent {
   onDismiss = () => {
     this.setState({modalVisibility: false}, () => {
       const props = this.getThemeProps();
-      _.invoke(props, 'onDismiss', props);
+      if (props.visible) {
+        _.invoke(props, 'onDismiss', props);
+      }
+      // Parity with iOS Modal's onDismiss
+      if (Constants.isAndroid) {
+        _.invoke(props, 'onModalDismissed', props);
```

Comment:
I feel this is risky, a user could have counted on this *not* working on Android, and now it will, this is a big potential for bugs.

Reply:
as discussed, I added a new prop - `onDialogDismissed`